packages/core/src/util/txInspector.ts

```diff
@@ -159,56 +191,149 @@ export const valueReceivedInspector: TotalAddressInputsValueInspector = (ownAddr
+export const delegationInspector: DelegationInspector = certificateInspector<DelegationInspection>(
+  CertificateType.StakeDelegation
+);
+
+/**
+ * Inspects a transaction for a stake key deregistration certificate.
+ *
+ * @param {TxAlonzo} tx transaction to inspect
+ * @returns {StakeKeyRegistrationInspection} array of stake key deregistration certificates
+ */
+export const stakeKeyDeregistrationInspector: StakeKeyRegistrationInspector =
+  certificateInspector<StakeKeyRegistrationInspection>(CertificateType.StakeKeyDeregistration);
 
 /**
  * Inspects a transaction for a stake key registration certificate.
  *
  * @param {TxAlonzo} tx transaction to inspect
  * @returns {StakeKeyRegistrationInspection} array of stake key registration certificates
  */
-export const stakeKeyRegistrationInspector: StakeKeyRegistrationInspector = (tx: TxAlonzo) =>
-  (tx.body.certificates?.filter(
-    (cert) => cert.__typename === CertificateType.StakeKeyRegistration
-  ) as StakeAddressCertificate[]) ?? [];
+export const stakeKeyRegistrationInspector: StakeKeyRegistrationInspector =
+  certificateInspector<StakeKeyRegistrationInspection>(CertificateType.StakeKeyRegistration);
 
 /**
- * Inspects a transaction for a stake key deregistration certificate.
+ * Inspects a transaction for pool registration certificates.
  *
- * @param {TxAlonzo} tx transaction to inspect
- * @returns {StakeKeyRegistrationInspection} array of stake key deregistration certificates
+ * @param {TxAlonzo} tx transaction to inspect.
+ * @returns {PoolRegistrationInspection} array of pool registration certificates.
  */
-export const stakeKeyDeregistrationInspector: StakeKeyRegistrationInspector = (tx: TxAlonzo) =>
-  (tx.body.certificates?.filter(
-    (cert) => cert.__typename === CertificateType.StakeKeyDeregistration
-  ) as StakeAddressCertificate[]) ?? [];
+export const poolRegistrationInspector: PoolRegistrationInspector = certificateInspector<PoolRegistrationInspection>(
+  CertificateType.PoolRegistration
+);
+
+/**
+ * Inspects a transaction for pool retirement certificates.
+ *
+ * @param {TxAlonzo} tx transaction to inspect.
+ * @returns {PoolRetirementInspection} array of pool retirement certificates.
+ */
+export const poolRetirementInspector: PoolRetirementInspector = certificateInspector<PoolRetirementInspection>(
+  CertificateType.PoolRetirement
+);
 
 /**
  * Inspects a transaction for withdrawals.
  *
  * @param {TxAlonzo} tx transaction to inspect
  * @returns {WithdrawalInspection} accumulated withdrawal quantities
  */
-export const withdrawalInspector: WithdrawalInspector = (tx: TxAlonzo) =>
+export const withdrawalInspector: WithdrawalInspector = (tx) =>
   tx.body.withdrawals?.length ? BigIntMath.sum(tx.body.withdrawals.map(({ quantity }) => quantity)) : 0n;
 
+/**
+ * Inspects a transaction for minted or burned assets.
+ *
+ * @param {TxAlonzo} tx transaction to inspect.
+ * @returns {Value} A collection of minted and burned assets grouped by policy Id.
```

Comment:
`Value` is a type of `{coins, assets}`. Type can be inferred when using typescript.
```suggestion
 * @returns collections of minted and burned assets grouped by policy Id.
```